pkg/cmd/release/create/gitResources.go

```diff
@@ -0,0 +1,301 @@
+		if err != nil {
+			continue
+		}
+
+		overriddenGitResourceGitRefs = append(overriddenGitResourceGitRefs, resolvedOverride)
+	}
+
+	//now merge the parsed overrides with the baseline resources, applying any overrides that match
+	gitResourceGitRefsWithOverrides := ApplyGitResourceOverrides(gitResourcesBaseline, overriddenGitResourceGitRefs)
+
+outerLoop:
+	for {
+		err := printGitResourceGitRefs(stdout, gitResourceGitRefsWithOverrides)
+		if err != nil {
+			return nil, err
+		}
+
+		var resolvedOverride *GitResourceGitRef = nil
+		var answer = ""
+		err = asker(&survey.Input{
+			Message: GitResourceOverrideQuestion,
+		}, &answer, survey.WithValidator(func(ans interface{}) error {
+			str, ok := ans.(string)
+			if !ok {
+				return errors.New("internal error; answer was not a string")
+			}
+
+			switch str {
+			// valid response for continuing the loop; don't attempt to validate these
+			case "y", "u", "r", "?", "":
+				return nil
+			}
+
+			parsedOverride, err := ParseGitResourceGitRefString(str)
+			if err != nil {
+				return err
+			}
+
+			resolvedOverride, err = ResolveGitResourceOverride(parsedOverride, gitResourcesBaseline)
+			if err != nil {
+				return err
+			}
+
+			return nil //all good!
+		}))
+
+		// if validators return an error, survey retries itself; the errors don't end up at this level.
+		if err != nil {
+			return nil, err
+		}
+
+		switch answer {
+		case "y": // YES these are the git resources they want
+			break outerLoop
+		case "?": // help text
+			_, _ = fmt.Fprintf(stdout, output.FormatDoc(gitResourceGitRefLoopHelpText))
+		case "u": // undo!
+			if len(overriddenGitResourceGitRefs) > 0 {
+				//strip the last override from the list
+				overriddenGitResourceGitRefs = overriddenGitResourceGitRefs[:len(overriddenGitResourceGitRefs)-1]
+				gitResourceGitRefsWithOverrides = ApplyGitResourceOverrides(gitResourcesBaseline, overriddenGitResourceGitRefs)
+			} else {
+				_ = fmt.Sprint(stdout, "nothing to undo")
+			}
+
+		case "r": // reset! All the way back to the calculated versions, discarding even the stuff that came in from the cmdline
+			overriddenGitResourceGitRefs = make([]*GitResourceGitRef, 0)
+			gitResourceGitRefsWithOverrides = ApplyGitResourceOverrides(gitResourcesBaseline, overriddenGitResourceGitRefs) //applying any empty array gives us the original list
+		default:
+			if resolvedOverride != nil {
+				overriddenGitResourceGitRefs = append(overriddenGitResourceGitRefs, resolvedOverride)
+				gitResourceGitRefsWithOverrides = ApplyGitResourceOverrides(gitResourcesBaseline, overriddenGitResourceGitRefs)
+			}
+		}
+		// loop around and let them put in more input
+	}
+	return overriddenGitResourceGitRefs, nil
+}
+
+func ParseGitResourceGitRefString(gitResourceRef string) (*GitResourceGitRef, error) {
+	if strings.TrimSpace(gitResourceRef) == "" {
+		return nil, errors.New("empty git resource git ref specification")
```

Comment:
We just call them `git resource ref` in the help output.  TBH I'm not really sure which we should use.

Reply:
I standardized on `git resource git reference`